spec/models/event_spec.rb

```diff
@@ -15,8 +15,45 @@
 
 describe Event do
 
+  let(:event) { FactoryGirl.create :event, :with_two_date_ranges }
+
   it "is created by event factory" do
-    event = FactoryGirl.build(:event)
     expect(event).to be_valid
   end
+
+  it "should have one or more date-ranges" do
+
+    #checking if the event model can handle date_ranges
+    expect(event.date_ranges.size).to eq 2
+    expect(event.date_ranges.first.start_date).to eq(Date.tomorrow)
+    expect(event.date_ranges.first.end_date).to eq(Date.tomorrow.next_day(5))
+    expect(event.date_ranges.second.start_date).to eq(Date.tomorrow)
+    expect(event.date_ranges.second.end_date).to eq(Date.tomorrow.next_day(10))
+    expect(event.date_ranges.second).to eq(event.date_ranges.last)
+
+    #making sure that every event has at least one date range
+    event1 = FactoryGirl.build(:event, :without_date_ranges)
+    expect(event1).to_not be_valid
+  end
+
+  describe "#start_date" do
+    it "should return return its minimum over all date ranges" do
+      event = FactoryGirl.create :event, :with_multiple_date_ranges
+      expect(event.start_date).to eq(Date.today)
+    end
+  end
+
+  describe "#end_date" do
+    it "should return return its maximum over all date ranges" do
+      event = FactoryGirl.create :event, :with_multiple_date_ranges
+      expect(event.end_date).to eq(Date.today.next_day(16))
```

Comment:
It's derived from the definitions in the factory. Are we expected to handle all of these values as constants? Since it's both test data, we duplicated values in both places all the time. Otherwise, we would need to carry around about ten constants to get meaningful tests.